src/compat/libc/string/memccpy.c

```diff
@@ -0,0 +1,95 @@
+/**
+ * @file
+ * @brief Implementation of #memccpy() function.
+ * 
+ * @date 18/3/20
+ * @author Giannis Zapantis (github name: tech-gian)
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+
+void* memccpy(void* dest, const void* src, int c, size_t n) {
+
+/* If overlap happens the results
+ * will be undefined.
+ */
+
+
+/* if there is already data after c
+ * then that data stays the same.
+ * if there isn't we don't change
+ * anything.
+ */
+
+/* copying dest to temp_dest
+ * and src to temp_dest.
+ * So we can work for each
+ * byte separately.
+*/
+
+    char* temp_dest = (char*) dest;
+    char* temp_src = (char*) src;
+    size_t pos = n;
+
+    char* ar1[n];
+
+/* getting each byte for c. */
+    char temp_c[4];
+    temp_c[0] = (c >> 24) & 0xFF;
+    temp_c[1] = (c >> 16) & 0xFF;
+    temp_c[2] = (c >> 8) & 0xFF;
+    temp_c[3] = c & 0xFF;

```

Comment:
Are you sure there are 4 chars?

```
The  memccpy()  function  copies  no more than n bytes from memory area
src to memory area dest, stopping when the character c is found.
```
So `c` is just a character, not 4 characters.

Reply:
If we correct this the implementation could be much simpler.

Reply:
Below that I am checking only one byte (temp_c[3]), because as you said, it's only one char. But I think that because c is int, we have to check all 4 bytes, considering user's wrong input. The problem came up when I tested with ints instead of char* and then if I sent c=0, it gave me wrong results.